src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -21,255 +169,1401 @@ const ActorInit En_Nb_InitVars = {
+        }
+    }
+}
+
+void func_80AB1284(EnNb* this, GlobalContext* globalCtx) {
+    func_8002E4B4(globalCtx, &this->actor, 75.0f, 30.0f, 30.0f, 4);
+}
+
+s32 EnNb_FrameUpdateMatrix(EnNb* this) {
+    return SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+CsCmdActorAction* EnNb_GetNpcCsAction(GlobalContext* globalCtx, s32 npcActionIdx) {
+    if (globalCtx->csCtx.state != 0) {
+        return globalCtx->csCtx.npcActions[npcActionIdx];
+    }
+    return NULL;
+}
+
+void EnNb_SetupCsPosRot(EnNb* this, GlobalContext* globalCtx, s32 npcActionIdx) {
+    CsCmdActorAction* csCmdNPCAction = EnNb_GetNpcCsAction(globalCtx, npcActionIdx);
+    s16 newRotY;
+    Actor* thisx = &this->actor;
+
+    if (csCmdNPCAction != NULL) {
+        thisx->posRot.pos.x = csCmdNPCAction->startPos.x;
+        thisx->posRot.pos.y = csCmdNPCAction->startPos.y;
+        thisx->posRot.pos.z = csCmdNPCAction->startPos.z;
+        newRotY = csCmdNPCAction->rot.y;
+        thisx->shape.rot.y = newRotY;
+        thisx->posRot.rot.y = newRotY;
+    }
+}
+
+s32 func_80AB1390(EnNb* this, GlobalContext* globalCtx, u16 arg2, s32 npcActionIdx) {
+    CsCmdActorAction* csCmdNPCAction;
+    if ((globalCtx->csCtx.state != 0) &&
+        (csCmdNPCAction = globalCtx->csCtx.npcActions[npcActionIdx], csCmdNPCAction != NULL) &&
+        (csCmdNPCAction->action == arg2)) {
+        return 1;
+    }
+    return 0;
+}
+
+s32 func_80AB13D8(EnNb* this, GlobalContext* globalCtx, u16 arg2, s32 npcActionIdx) {
+    CsCmdActorAction* csCmdNPCAction;
+    if ((globalCtx->csCtx.state != 0) &&
+        (csCmdNPCAction = globalCtx->csCtx.npcActions[npcActionIdx], csCmdNPCAction != NULL) &&
+        (csCmdNPCAction->action != arg2)) {
+        return 1;
+    }
+    return 0;
+}
+
+void EnNb_SetInitialCsPosRot(EnNb* this, GlobalContext* globalCtx, s32 npcActionIdx) {
+    CsCmdActorAction* csCmdNPCAction = EnNb_GetNpcCsAction(globalCtx, npcActionIdx);
+    s16 newRotY;
+    Actor* thisx = &this->actor;
+
+    if (csCmdNPCAction != NULL) {
+        thisx->posRot.pos.x = csCmdNPCAction->startPos.x;
+        thisx->posRot.pos.y = csCmdNPCAction->startPos.y;
+        thisx->posRot.pos.z = csCmdNPCAction->startPos.z;
+        newRotY = csCmdNPCAction->rot.y;
+        thisx->shape.rot.y = newRotY;
+        thisx->posRot.rot.y = newRotY;
+    }
+}
+
+void EnNb_SetCurrentAnim(EnNb* this, AnimationHeader* animation, u8 mode, f32 transitionRate, s32 arg4) {
+    f32 frameCount = SkelAnime_GetFrameCount(&animation->genericHeader);
+    f32 playbackSpeed;
+    f32 unk0;
+    f32 fc;
+
+    if (!arg4) {
+        unk0 = 0.0f;
+        fc = frameCount;
+        playbackSpeed = 1.0f;
+    } else {
+        unk0 = frameCount;
+        fc = 0.0f;
+        playbackSpeed = -1.0f;
+    }
+
+    SkelAnime_ChangeAnim(&this->skelAnime, animation, playbackSpeed, unk0, fc, mode, transitionRate);
+}
+
+void EnNb_SetCurrentAnim(EnNb* this, AnimationHeader* animation, u8 mode, f32 transitionRate, s32 arg4);
```

Comment:
```suggestion
```
useless prototype

even its prototype at the top of the file is useless, the function being defined before it's used (may be the case for other functions too)